src/Swift.Runtime/src/TypeDatabase.cs

```diff
@@ -0,0 +1,123 @@
+
+            XmlNode? entitiesNode = xmlDoc?.SelectSingleNode("//swifttypedatabase/entities");
+            if (entitiesNode == null)
+                return false;
+
+            if (entitiesNode.ChildNodes.Count == 0)
+                return false;
+
+            foreach (XmlNode entityNode in entitiesNode.ChildNodes)
+            {
+                if (entityNode.Name != "entity")
+                    return false;
+
+                XmlNode? typeDeclarationNode = entityNode?.SelectSingleNode("typedeclaration");
+                if (typeDeclarationNode == null)
+                    return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Reads and parses the XML document containing type mappings based on the version 1.0.
+        /// </summary>
+        /// <param name="xmlDoc">The XML document to read.</param>
+        private void ReadVersion1_0(XmlDocument xmlDoc)
+        {
+            XmlNode? entitiesNode = xmlDoc.SelectSingleNode("//swifttypedatabase/entities");
+
+            if (entitiesNode == null)
+            	throw new Exception("Invalid XML structure: 'entities' node not found.");
+
+            foreach (XmlNode? entityNode in entitiesNode.ChildNodes)
+            {
+                XmlNode? typeDeclarationNode = entityNode?.SelectSingleNode("typedeclaration");
+                if (typeDeclarationNode == null)
+                    throw new Exception("Invalid XML structure: 'typedeclaration' node not found.");
+                    
+                string? swiftName = typeDeclarationNode?.Attributes?["name"]?.Value;
+                string? csharpName = entityNode?.Attributes?["managedTypeName"]?.Value;
+                if (swiftName == null || csharpName == null)
+                    throw new Exception("Invalid XML structure: Missing attributes.");
+
+                _swiftToCSharpMapping.Add(swiftName, csharpName);
+            }
+        }
+
+        /// <summary>
+        /// Gets the C# type name corresponding to the specified Swift type name.
+        /// </summary>
+        /// <param name="swiftName">The Swift type name.</param>
+        /// <returns>The corresponding C# type name.</returns>
+        public string GetCSharpName(string swiftName)
+        {
+            if (_swiftToCSharpMapping.TryGetValue(swiftName, out string? csharpName))
+            {
+                return csharpName;
+            }
+            throw new Exception($"No mapping for {swiftName} found in the type database.");
+        }
+
+        /// <summary>
+        /// Checks if a Swift type name corresponds to a known C# value type.
+        /// </summary>
+        /// <param name="swiftName">The Swift type name.</param>
+        /// <returns><c>true</c> if the Swift type name maps to a C# value type; otherwise <c>false</c>.</returns>
+        public bool IsValueType(string swiftName)
+        {
+            return _swiftToCSharpMapping.TryGetValue(swiftName, out string? csharpName);
```

Comment:
Do we expect the mapping in `src/Swift.Runtime/src/TypeDatabase.xml` to grow in the future so that it would contain also other types that don't match to C# value types? For example, strings?

Reply:
The `TypeDatabase.xml` should provide mappings for common C# types. It can be expanded in the future, and we need to project each type pattern individually. For non-trivial mappings, a projection will be created within the `Swift.Runtime` namespace (i.e. unsafe buffer pointers). Currently, the type database attempts to find a known mapping. If not found, the type is then searched for within `Swift.Runtime` namespace:
```csharp
/// <summary>
/// Gets the C# type name corresponding to the specified Swift type name. 
/// The method first tries to find a known mapping, and if that fails, it looks for a type in Swift.Runtime.
/// </summary>
/// <param name="swiftTypeName">The Swift type name.</param>
/// <returns>The corresponding C# type name.</returns>
public string[] GetCSharpTypeName(string swiftTypeName)
{
    // Try to find a known mapping
    if (_swiftToCSharpMapping.TryGetValue(swiftTypeName, out string? csharpTypeName))
    {
        return new string [] { "System", csharpTypeName };
    }

    // Try to find a type in Swift.Runtime
    Type? swiftRuntimeType = Type.GetType($"Swift.Runtime.{swiftTypeName}");
    if (swiftRuntimeType != null)
        return new string [] { "Swift.Runtime", swiftRuntimeType.Name };
    
    throw new Exception($"No mapping for {swiftTypeName} type found.");
}
```

Reply:
Thank you for the explanation